src/core/electrostatics_magnetostatics/scafacos.cpp

```diff
@@ -113,43 +113,44 @@ void ScafacosData::update_particle_forces() const {
 
   for (auto &p : local_cells.particles()) {
     if (!dipolar()) {
-      p.f.f[0] += coulomb.prefactor * p.p.q * fields[it++];
-      p.f.f[1] += coulomb.prefactor * p.p.q * fields[it++];
-      p.f.f[2] += coulomb.prefactor * p.p.q * fields[it++];
+      p.f.f += coulomb.prefactor * p.p.q *
+               Utils::Vector3d(Utils::Span<const double>(&(fields[it]), 3));
+      it += 3;
     } else {
 #ifdef SCAFACOS_DIPOLES
       // Indices
       // 3 "potential" values per particles (see below)
-      int it_p = 3 * it;
+      int const it_p = 3 * it;
       // 6 "field" values per particles (see below)
-      int it_f = 6 * it;
+      int const it_f = 6 * it;
 
       // The scafacos term "potential" here in fact refers to the magnetic
       // field
       // So, the torques are given by m \times B
-      const Vector3d dip = p.calc_dip();
-      auto const t = dip.cross(
-          Vector3d(Utils::Span<const double>(&(potentials[it_p]), 3)));
+      auto const dip = p.calc_dip();
+      auto const t = vector_product(
+          dip,
+          Utils::Vector3d(Utils::Span<const double>(&(potentials[it_p]), 3)));
       // The force is given by G m, where G is a matrix
       // which comes from the "fields" output of scafacos like this
       // 0 1 2
       // 1 3 4
       // 2 4 5
-      // where the numbers refer to indices in the "field" output from
-      // scafacos
-      double f[3];
-      f[0] = fields[it_f + 0] * dip[0] + fields[it_f + 1] * dip[1] +
-             fields[it_f + 2] * dip[2];
-      f[1] = fields[it_f + 1] * dip[0] + fields[it_f + 3] * dip[1] +
-             fields[it_f + 4] * dip[2];
-      f[2] = fields[it_f + 2] * dip[0] + fields[it_f + 4] * dip[1] +
-             fields[it_f + 5] * dip[2];
+      // where the numbers refer to indices in the "field" output from scafacos
+      Utils::Vector3d const f = {
```

Comment:
Wouldn't it be clearer to just construct the matrix and then do a product? like
```
auto const G = Vector<Vector3d, 3>>{{fields[it_f + 0], ...}, ...};
auto const f = G * dip;
```
?

Reply:
Thanks! This is a lot better.